lib/os_io.c

```diff
@@ -37,6 +37,199 @@ ___io_module ___io_mod =
 
 /*---------------------------------------------------------------------------*/
 
+/* poll dynamic fdset memory management. */
+
+#ifdef USE_select_or_poll
+
+___HIDDEN int ___fdset_heap_overflow;
```

Comment:
Agreed about the flag being in vmstate, will move it there; not sure about the actual fdset size though, as it might differ temporarily between processors while the resizing is in progress.